pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4514,4 +4514,36 @@ private void internalSetReplicatedSubscriptionStatusForNonPartitionedTopic(Async
             resumeAsyncResponseExceptionally(asyncResponse, e);
         }
     }
+
+    protected Map<String, Boolean> internalGetReplicatedSubscriptionStatus() {
+        log.info("[{}] Attempting to get replicated subscription status on {}", clientAppId(), topicName);
+
+        // Reject the request if the topic is not persistent
+        if (!topicName.isPersistent()) {
+            new RestException(Status.METHOD_NOT_ALLOWED,
+                    "Cannot get replicated subscriptions on non-persistent topics");
+        }
+
+        // Reject the request if the topic is not global
+        if (!topicName.isGlobal()) {
+            new RestException(Status.METHOD_NOT_ALLOWED,
+                    "Cannot get replicated subscriptions on non-global topics");
+        }
+
+        // Permission to consume this topic is required
+        validateTopicOperation(topicName, TopicOperation.GET_REPLICATED_SUBSCRIPTION_STATUS);
+
+        Topic topic = getTopicReference(topicName);
```

Comment:
Need to check the ownership first, otherwise the topic might be owned by another broker.

Reply:
And the partitioned topic should be handled.

Reply:
Thanks for your sugesstion and I have add related logic, PTAL again :)